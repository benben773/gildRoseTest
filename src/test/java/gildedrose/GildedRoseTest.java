package gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class GildedRoseTest {

    @Test
    public void approveTest() {

        CombinationApprovals.verifyAllCombinations(
                this::getString,
                new String[]{"foo","Aged Brie","Sulfuras, Hand of Ragnaros","Backstage passes to a TAFKAL80ETC concert"},
                new Integer[]{-1,0,1,10,11,12},
                new Integer[]{-1,0,1,49,50,51}
        );
    }

    private String getString(String foo, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(foo, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }

}
