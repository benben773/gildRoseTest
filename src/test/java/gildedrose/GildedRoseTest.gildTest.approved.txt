[foo, 0, 0] => foo, -1, 0 
[foo, 0, 10] => foo, -1, 8 
[foo, 0, 12] => foo, -1, 10 
[foo, 0, -2] => foo, -1, -2 
[foo, 0, 51] => foo, -1, 49 
[foo, 0, 78] => foo, -1, 76 
[foo, 0, 49] => foo, -1, 47 
[foo, 1, 0] => foo, 0, 0 
[foo, 1, 10] => foo, 0, 9 
[foo, 1, 12] => foo, 0, 11 
[foo, 1, -2] => foo, 0, -2 
[foo, 1, 51] => foo, 0, 50 
[foo, 1, 78] => foo, 0, 77 
[foo, 1, 49] => foo, 0, 48 
[foo, 10, 0] => foo, 9, 0 
[foo, 10, 10] => foo, 9, 9 
[foo, 10, 12] => foo, 9, 11 
[foo, 10, -2] => foo, 9, -2 
[foo, 10, 51] => foo, 9, 50 
[foo, 10, 78] => foo, 9, 77 
[foo, 10, 49] => foo, 9, 48 
[foo, -1, 0] => foo, -2, 0 
[foo, -1, 10] => foo, -2, 8 
[foo, -1, 12] => foo, -2, 10 
[foo, -1, -2] => foo, -2, -2 
[foo, -1, 51] => foo, -2, 49 
[foo, -1, 78] => foo, -2, 76 
[foo, -1, 49] => foo, -2, 47 
[foo, 12, 0] => foo, 11, 0 
[foo, 12, 10] => foo, 11, 9 
[foo, 12, 12] => foo, 11, 11 
[foo, 12, -2] => foo, 11, -2 
[foo, 12, 51] => foo, 11, 50 
[foo, 12, 78] => foo, 11, 77 
[foo, 12, 49] => foo, 11, 48 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 10] => Aged Brie, -1, 12 
[Aged Brie, 0, 12] => Aged Brie, -1, 14 
[Aged Brie, 0, -2] => Aged Brie, -1, 0 
[Aged Brie, 0, 51] => Aged Brie, -1, 51 
[Aged Brie, 0, 78] => Aged Brie, -1, 78 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 10] => Aged Brie, 0, 11 
[Aged Brie, 1, 12] => Aged Brie, 0, 13 
[Aged Brie, 1, -2] => Aged Brie, 0, -1 
[Aged Brie, 1, 51] => Aged Brie, 0, 51 
[Aged Brie, 1, 78] => Aged Brie, 0, 78 
[Aged Brie, 1, 49] => Aged Brie, 0, 50 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 10] => Aged Brie, 9, 11 
[Aged Brie, 10, 12] => Aged Brie, 9, 13 
[Aged Brie, 10, -2] => Aged Brie, 9, -1 
[Aged Brie, 10, 51] => Aged Brie, 9, 51 
[Aged Brie, 10, 78] => Aged Brie, 9, 78 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 10] => Aged Brie, -2, 12 
[Aged Brie, -1, 12] => Aged Brie, -2, 14 
[Aged Brie, -1, -2] => Aged Brie, -2, 0 
[Aged Brie, -1, 51] => Aged Brie, -2, 51 
[Aged Brie, -1, 78] => Aged Brie, -2, 78 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, 12, 0] => Aged Brie, 11, 1 
[Aged Brie, 12, 10] => Aged Brie, 11, 11 
[Aged Brie, 12, 12] => Aged Brie, 11, 13 
[Aged Brie, 12, -2] => Aged Brie, 11, -1 
[Aged Brie, 12, 51] => Aged Brie, 11, 51 
[Aged Brie, 12, 78] => Aged Brie, 11, 78 
[Aged Brie, 12, 49] => Aged Brie, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 10] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 12] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, -2] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 51] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 78] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 10] => Backstage passes to a TAFKAL80ETC concert, 0, 13 
[Backstage passes to a TAFKAL80ETC concert, 1, 12] => Backstage passes to a TAFKAL80ETC concert, 0, 15 
[Backstage passes to a TAFKAL80ETC concert, 1, -2] => Backstage passes to a TAFKAL80ETC concert, 0, 1 
[Backstage passes to a TAFKAL80ETC concert, 1, 51] => Backstage passes to a TAFKAL80ETC concert, 0, 51 
[Backstage passes to a TAFKAL80ETC concert, 1, 78] => Backstage passes to a TAFKAL80ETC concert, 0, 78 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 10] => Backstage passes to a TAFKAL80ETC concert, 9, 12 
[Backstage passes to a TAFKAL80ETC concert, 10, 12] => Backstage passes to a TAFKAL80ETC concert, 9, 14 
[Backstage passes to a TAFKAL80ETC concert, 10, -2] => Backstage passes to a TAFKAL80ETC concert, 9, 0 
[Backstage passes to a TAFKAL80ETC concert, 10, 51] => Backstage passes to a TAFKAL80ETC concert, 9, 51 
[Backstage passes to a TAFKAL80ETC concert, 10, 78] => Backstage passes to a TAFKAL80ETC concert, 9, 78 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 10] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 12] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, -2] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 51] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 78] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Backstage passes to a TAFKAL80ETC concert, 11, 1 
[Backstage passes to a TAFKAL80ETC concert, 12, 10] => Backstage passes to a TAFKAL80ETC concert, 11, 11 
[Backstage passes to a TAFKAL80ETC concert, 12, 12] => Backstage passes to a TAFKAL80ETC concert, 11, 13 
[Backstage passes to a TAFKAL80ETC concert, 12, -2] => Backstage passes to a TAFKAL80ETC concert, 11, -1 
[Backstage passes to a TAFKAL80ETC concert, 12, 51] => Backstage passes to a TAFKAL80ETC concert, 11, 51 
[Backstage passes to a TAFKAL80ETC concert, 12, 78] => Backstage passes to a TAFKAL80ETC concert, 11, 78 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros, 0, 10 
[Sulfuras, Hand of Ragnaros, 0, 12] => Sulfuras, Hand of Ragnaros, 0, 12 
[Sulfuras, Hand of Ragnaros, 0, -2] => Sulfuras, Hand of Ragnaros, 0, -2 
[Sulfuras, Hand of Ragnaros, 0, 51] => Sulfuras, Hand of Ragnaros, 0, 51 
[Sulfuras, Hand of Ragnaros, 0, 78] => Sulfuras, Hand of Ragnaros, 0, 78 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros, 1, 0 
[Sulfuras, Hand of Ragnaros, 1, 10] => Sulfuras, Hand of Ragnaros, 1, 10 
[Sulfuras, Hand of Ragnaros, 1, 12] => Sulfuras, Hand of Ragnaros, 1, 12 
[Sulfuras, Hand of Ragnaros, 1, -2] => Sulfuras, Hand of Ragnaros, 1, -2 
[Sulfuras, Hand of Ragnaros, 1, 51] => Sulfuras, Hand of Ragnaros, 1, 51 
[Sulfuras, Hand of Ragnaros, 1, 78] => Sulfuras, Hand of Ragnaros, 1, 78 
[Sulfuras, Hand of Ragnaros, 1, 49] => Sulfuras, Hand of Ragnaros, 1, 49 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 10] => Sulfuras, Hand of Ragnaros, 10, 10 
[Sulfuras, Hand of Ragnaros, 10, 12] => Sulfuras, Hand of Ragnaros, 10, 12 
[Sulfuras, Hand of Ragnaros, 10, -2] => Sulfuras, Hand of Ragnaros, 10, -2 
[Sulfuras, Hand of Ragnaros, 10, 51] => Sulfuras, Hand of Ragnaros, 10, 51 
[Sulfuras, Hand of Ragnaros, 10, 78] => Sulfuras, Hand of Ragnaros, 10, 78 
[Sulfuras, Hand of Ragnaros, 10, 49] => Sulfuras, Hand of Ragnaros, 10, 49 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 10] => Sulfuras, Hand of Ragnaros, -1, 10 
[Sulfuras, Hand of Ragnaros, -1, 12] => Sulfuras, Hand of Ragnaros, -1, 12 
[Sulfuras, Hand of Ragnaros, -1, -2] => Sulfuras, Hand of Ragnaros, -1, -2 
[Sulfuras, Hand of Ragnaros, -1, 51] => Sulfuras, Hand of Ragnaros, -1, 51 
[Sulfuras, Hand of Ragnaros, -1, 78] => Sulfuras, Hand of Ragnaros, -1, 78 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, 12, 0] => Sulfuras, Hand of Ragnaros, 12, 0 
[Sulfuras, Hand of Ragnaros, 12, 10] => Sulfuras, Hand of Ragnaros, 12, 10 
[Sulfuras, Hand of Ragnaros, 12, 12] => Sulfuras, Hand of Ragnaros, 12, 12 
[Sulfuras, Hand of Ragnaros, 12, -2] => Sulfuras, Hand of Ragnaros, 12, -2 
[Sulfuras, Hand of Ragnaros, 12, 51] => Sulfuras, Hand of Ragnaros, 12, 51 
[Sulfuras, Hand of Ragnaros, 12, 78] => Sulfuras, Hand of Ragnaros, 12, 78 
[Sulfuras, Hand of Ragnaros, 12, 49] => Sulfuras, Hand of Ragnaros, 12, 49 
